tests/test_schemas.py

```diff
@@ -172,22 +216,274 @@ def test_sub_schema_graph(self):
+        
+        # In the current example model, there are no properties, would need to update this section if properties are added.
+        assert data_model_properties == []
+
+        # Update the attr_rel_dictionary to add a property, then see if its found.
+        # Get a random relationship key from the attr_rel_dictionary:
+        all_keys = list(attr_rel_dictionary.keys())
+        random_index = len(all_keys)-1
+        rel_key = all_keys[random.randint(0, random_index)]
+
+        # Modify the contents of that relationship
+        attr_rel_dictionary[rel_key]['Relationships']['Properties'] = ['TestProperty']
+        
+        # Get properties in the modified data model
+        data_model_properties = data_model_nodes.get_data_model_properties(attr_rel_dictionary)
+
+        assert data_model_properties == ['TestProperty']
+
+    @pytest.mark.parametrize("data_model", list(DATA_MODEL_DICT.keys()), ids=list(DATA_MODEL_DICT.values()))
+    def test_get_entry_type(self, helpers, data_model):
+        
+        # Instantiate Parser
+        data_model_parser = helpers.get_data_model_parser(data_model_name=data_model)
+
+        # Parse Model
+        attr_rel_dictionary = data_model_parser.parse_model()
+
+        # Update the attr_rel_dictionary to add a property, then see if it is assigned the correct entry type.
+        # Get a random relationship key from the attr_rel_dictionary:
+        all_keys = list(attr_rel_dictionary.keys())
+        random_index = len(all_keys)-1
+        rel_key = all_keys[random.randint(0, random_index)]
+
+        # Modify the contents of that relationship
+        attr_rel_dictionary[rel_key]['Relationships']['Properties'] = ['TestProperty']
+
+        # Instantiate DataModelNodes
+        # Note: Get entry type uses self, so I will have to instantiate DataModelNodes outside of the generate_data_model_nodes function
+        data_model_nodes = DataModelNodes(attribute_relationships_dict=attr_rel_dictionary)
+
+        # In the example data model all attributes should be classes.
+        for attr in attr_rel_dictionary.keys():
+            entry_type = data_model_nodes.get_entry_type(attr)
+            assert entry_type == 'class'
+
+        # Check that the added property is properly loaded as a property
+        assert data_model_nodes.get_entry_type('TestProperty') == 'property'
+
+    @pytest.mark.parametrize("data_model", list(DATA_MODEL_DICT.keys()), ids=list(DATA_MODEL_DICT.values()))
+    @pytest.mark.parametrize("rel_func", list(REL_FUNC_DICT.values()), ids=list(REL_FUNC_DICT.keys()))
+    @pytest.mark.parametrize("test_dn", list(TEST_DN_DICT.keys()), ids=list(TEST_DN_DICT.keys()))
+    @pytest.mark.parametrize("test_bool", ['True', 'False', True, False, 'kldjk'], ids=['True_str', 'False_str', 'True_bool', 'False_bool', 'Random_str'])
+    def test_run_rel_functions(self, helpers, data_model, rel_func, test_dn, test_bool):
```

Comment:
Do we have unit tests for `get_attribute_display_name_from_label`, `parse_validation_rules`, `get_label_from_display_name`, `convert_bool_to_str`? If so, I think here we could simply mock responses from these four functions and test `run_rel_functions`.. Currently what you are doing here makes sense. But at the same time, I feel like this test becomes difficult to read because it gathers essentially four tests of four different functions

Reply:
we do have tests for those, but the trick here is actually making sure the parameters are being passed properly through this function.. which has at times felt a bit chaotic, so its nice to call the functions this way.